mongo/mongo.go

```diff
@@ -0,0 +1,43 @@
+package mongo
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	mgo "gopkg.in/mgo.v2"
+)
+
+func Close(ctx context.Context, session *mgo.Session) error {
+	closedChannel := make(chan bool)
+	defer close(closedChannel)
+
+	timeLeft := 1000 * time.Millisecond
```

Comment:
Is it better to have the timeout as a parameter to the function? As it allows the user of the library to control it.

Reply:
Not really, if a user wants to control how the sessions close they should use context deadline. The timeLeft variable is a default value just in case a deadline is not set. By giving them control of timeLeft could lead a user to using this variable over the context deadline defeating  the purpose of context.

Instead i will set `timeLeft` variable to a constant.